src/silverlabnwb/header.py

```diff
@@ -216,8 +216,11 @@ def determine_trial_times(self):
         # (misleadingly titled) section of the header.
         trial_times = []
         number_of_trials = ceil(len(self['Intertrial FIFO Times']) / 2)
+        last_time_present = (len(self['Intertrial FIFO Times']) == 2 * number_of_trials)
         for i in range(number_of_trials):
-            trial_times.append(
-                (self['Intertrial FIFO Times'][2 * i], self['Intertrial FIFO Times'][2 * i + 1]))
-        # TODO handle last trial separately in case stop_time is missing
+            if i < (number_of_trials-1) or last_time_present:
+                start, end = self['Intertrial FIFO Times'][2 * i], self['Intertrial FIFO Times'][2 * i + 1]
+            else:
+                start, end = (self['Intertrial FIFO Times'][2 * i], None)  # determine final 'end' later from speed data
```

Comment:
It might be a little easier to read if we assign `start` first (same in both cases), then only assign `end` inside the `if ...`. But maybe not.
Also, possibly add a brief comment here about what we're doing/why this is necessary? (we may forget!)